src/callbacks/callbacks.jl

```diff
@@ -434,6 +434,64 @@ function save_restart_func(integrator, output_dir)
     return nothing
 end
 
+Base.@kwdef mutable struct WallTimeEstimate
+    n_calls::Int = 0
+    n_next::Int = 1
+    t_last::Float64 = -1
+    ∑Δt_wall::Float64 = 0
+end
+import Dates
+function print_walltime_estimate(integrator)
+    (; walltime_estimate, dt, t_end) = integrator.p
+    wte = walltime_estimate
+    # 1st call initializes `t_last` (during integrator init)
+    # 2nd call computes `Δt` for the first time (but also includes compilation), so we skip
+    cond = wte.n_calls > 1
+    if cond
+        # We need to account for skipping cost of `Δt` when `n_calls == 1`:
+        factor = wte.n_calls == 2 ? 2 : 1
+        Δt = factor * (time() - wte.t_last)
+    else
+        wte.n_calls == 1 && @info "Progress: Completed first step"
+        Δt = Float64(0)
+        wte.n_next = wte.n_calls + 1
+    end
+    wte.∑Δt_wall += Δt
+    wte.t_last = time()
+
+    if wte.n_calls == wte.n_next && cond
+        t = integrator.t
+        n_steps_total = ceil(Int, t_end / dt)
```

Comment:
I think that this might not be correct when we restart a simulation from a checkpoint.

Reply:
Yeah, it's not correct for restarted simulations. I'd like to extend support for this in a followup PR-- I think the logic will be a bit more complicated (I've done this before in another project, but in Fortran). IIRC, the reason it's not as simple is because we're accumulating `dt_wall` to provide an estimate that kind of converges over time (callbacks and variation in timings can make the results noisy, which is a downside of not accumulating the estimates). So, if we wanted to keep the accumulation version, we'd need to save this to disc and read it in for restarted simulations.

Reply:
For now, I'll just add a warning if a simulation is restarted.